batched/dense/impl/KokkosBatched_Trsm_Serial_Impl.hpp

```diff
@@ -320,25 +705,152 @@ struct SerialTrsm<Side::Left, Uplo::Upper, Trans::Transpose, ArgDiag, Algo::Trsm
+        B.stride_1(), B.stride_0());
+  }
+};
+
+// [TO DO] ConjTranspose is not supported yet
+
+///
+/// R/U/C
+///
+/// B := (alpha*B) inv(triu(AH))
+/// A(n x n), B(m x n)
+#if defined(KOKKOSBATCHED_IMPL_ENABLE_INTEL_MKL) && defined(KOKKOSBATCHED_IMPL_ENABLE_INTEL_MKL_BATCHED) && \
+    defined(__KOKKOSBATCHED_ENABLE_INTEL_MKL_COMPACT_BATCHED__)
+template <typename ArgDiag>
+struct SerialTrsm<Side::Right, Uplo::Upper, Trans::ConjTranspose, ArgDiag, Algo::Trsm::CompactMKL> {
+  template <typename ScalarType, typename AViewType, typename BViewType>
+  KOKKOS_INLINE_FUNCTION static int invoke(const ScalarType alpha, const AViewType &A, const BViewType &B) {
+    typedef typename BViewType::value_type vector_type;
+    // typedef typename vector_type::value_type value_type;
+
+    const int m = B.extent(0), n = B.extent(1);
+
+    static_assert(is_vector<vector_type>::value, "value type is not vector type");
+    static_assert(vector_type::vector_length == 4 || vector_type::vector_length == 8,
+                  "AVX, AVX2 and AVX512 is supported");
+    const MKL_COMPACT_PACK format = vector_type::vector_length == 8 ? MKL_COMPACT_AVX512 : MKL_COMPACT_AVX;
+
+    // no error check
+    int r_val = 0;
+    if (A.stride_0() == 1 && B.stride_0() == 1) {
+      mkl_dtrsm_compact(MKL_COL_MAJOR, MKL_RIGHT, MKL_UPPER, MKL_CONJTRANS,
+                        ArgDiag::use_unit_diag ? MKL_UNIT : MKL_NONUNIT, m, n, alpha, (const double *)A.data(),
+                        A.stride_1(), (double *)B.data(), B.stride_1(), format, (MKL_INT)vector_type::vector_length);
+    } else if (A.stride_1() == 1 && B.stride_1() == 1) {
+      mkl_dtrsm_compact(MKL_ROW_MAJOR, MKL_RIGHT, MKL_UPPER, MKL_CONJTRANS,
+                        ArgDiag::use_unit_diag ? MKL_UNIT : MKL_NONUNIT, m, n, alpha, (const double *)A.data(),
+                        A.stride_0(), (double *)B.data(), B.stride_0(), format, (MKL_INT)vector_type::vector_length);
+    } else {
+      r_val = -1;
+    }
+    return r_val;
+  }
+};
+#endif
+
+template <typename ArgDiag>
+struct SerialTrsm<Side::Right, Uplo::Upper, Trans::ConjTranspose, ArgDiag, Algo::Trsm::Unblocked> {
   template <typename ScalarType, typename AViewType, typename BViewType>
   KOKKOS_INLINE_FUNCTION static int invoke(const ScalarType alpha, const AViewType &A, const BViewType &B) {
-    return SerialTrsmInternalLeftLower<Algo::Trsm::Blocked>::invoke(ArgDiag::use_unit_diag, B.extent(0), B.extent(1),
-                                                                    alpha, A.data(), A.stride_1(), A.stride_0(),
-                                                                    B.data(), B.stride_0(), B.stride_1());
+    // Quick return if possible
+    if (B.extent(0) == 0 || B.extent(1) == 0) return 0;
+
+    auto info = KokkosBatched::Impl::checkTrsmInput<Side::Right>(A, B);
+    if (info) return info;
+
+    return KokkosBatched::Impl::SerialTrsmInternalLeftUpper<Algo::Trsm::Unblocked>::invoke(
```

Comment:
I assume you are doing: T^tX=B <=> X^tT=B^t so you can solve with T on the right and no transpose?

Reply:
If your `T` represents `A` in the code, yes, it is what we are doing here.